sirepo/package_data/static/js/sirepo.js

```diff
@@ -1246,15 +1254,49 @@ SIREPO.app.factory('frameCache', function(appState, panelState, requestSender, $
         if (! appState.isLoaded()) {
             return;
         }
-        function onError() {
-            panelState.reportNotGenerated(modelName);
+
+        let isHidden = panelState.isHidden(modelName);
+        let frameRequestTime = now();
+        let setPanelStateIsLoadingTimer = null;
+	const computeJobSerial = getComputeJobSerial();
+
+	function cancelSetPanelStateIsLoadingTimer() {
+	    if (setPanelStateIsLoadingTimer && computeJobSerialIsCurrent()) {
+		$timeout.cancel(setPanelStateIsLoadingTimer);
+		setPanelStateIsLoadingTimer = null;
+	    }
+	}
+
+	function computeJobSerialIsCurrent() {
+	    //Validate whether the current computeJobSerial is the
+	    //same as the one the frame request was for
+	    if (! appState.isLoaded()) {
+		return false;
+	    }
+	    return computeJobSerial === getComputeJobSerial();
```

Comment:
By param hash do you mean compute job hash? If so, it isn't enough. If a user doesn't change any parameters but clicks run simulation again then the hash is unchanged. But we don't want to show the frames because they are for an old run of the simulation. Furthermore, we don't generate a hash for [parallel simulations](https://github.com/radiasoft/sirepo/blob/5eecebd2b47618af00216849dfaf0488d781b20a/sirepo/sim_data/__init__.py#L203).

 Maybe we need a new concept? Some unique identifier of a single run of a simulation?

Reply:
We should generate a hash for parallel simulations. 

>  clicks run simulation again then the hash is unchanged. But we don't want to show the frames because they are for an old run of the simulation. 

The click should be better integrated in the supervisor. You click run, the graphs disappear (denormalized) in the UI. When a new browser visits a simulation (the serial and hash will be "up to date") that is currently running, frames won't be shown, because the frames will not be valid until they are available.

Reply:
What does the supervisor have to do with this? As I see it the code is protecting itself against responses for data that is no longer relevant. I believe the GUI has to be the one in charge of this decision.  

One of the cases this code is handling is this issue you opened https://github.com/radiasoft/sirepo/issues/4367 which doesn't involve a new browser.

Reply:
getComputeJobSerial should protect itself against this being only partially defined `appState.models.simulationStatus[appState.appService.computeModel(modelName)].computeJobSerial;` when the function runs.

Reply:
https://github.com/radiasoft/sirepo/blob/115ebe804e973eb29373b3d463ae9c6a6c9e4508/sirepo/package_data/static/js/sirepo.js#L1333 is the incorrect call to onError.